lib/modules/dosomething/dosomething_reportback/dosomething_reportback.query.inc

```diff
@@ -55,6 +55,15 @@ function dosomething_reportback_build_reportbacks_query($params = array()) {
     }
   }
 
+  if (!isset($params['flagged'])) {
```

Comment:
The issue is that the `flagged` param is sometimes passed and sometimes not, like in the case for viewing a specific RB where you just pass an id (so the rbid is the only param in the array).

The issue is that as an admin, you can pass `true` OR `false`... in that scenario `isset()` always evaluates to true, since there is a value set... and the condition runs (even if `false` is passed).